src/App.tsx

```diff
@@ -165,38 +182,75 @@ class Board extends React.Component<BoardProps, BoardState> {
+        this.setState(newState);
+    }
+
+    public handleSquareMouseOver = (i: SquareAddress) => (e: MouseEvent) => {
+        e.preventDefault();
+        if (this.state.highlighting) {
+            console.log("SquareMouseOver at addy: " + i);
+            const newState = _.cloneDeep(this.state);
+            newState.highlights.add(i);
+            this.setState(newState);
+        }
+    }
+
+    public handleSquareMouseUp = (i: SquareAddress) => (e: MouseEvent) => {
+        e.preventDefault();
+        console.log("SquareMouseDown at SQaddy: " + i);
+        const newState = _.cloneDeep(this.state);
+        newState.highlighting = false;
+        this.setState(newState);
+    }
+
     public render() {
         return (
-            <div className="game" onKeyPress={this.handlePress} tabIndex={0}>
-                <div className="board">{this.renderSquares()}</div>
-                <div className="button-box">
-                    <div className="button-box-top">
-                        <ControlButtons
-                            onClickMode={() => {
-                                this.toggleNumpadMode();
-                            }}
-                            numpadMode={this.state.numpadMode}
-                        />
-                        <Numpad
-                            numpadMode={this.state.numpadMode}
-                            onClickCorner={(i: SquareValue) => {
-                                this.cornerMarkSelectedSquares(i);
-                            }}
-                            onClickNormal={(i: SquareValue) => {
-                                this.normalMarkSelectedSquares(i);
-                            }}
-                            onClickDel={() => {
-                                this.deleteSelectedSquares();
-                            }}
-                        />
-                    </div>
-                    <div className="button-box-bot">
-                        <Button variant="contained" color="primary">
-                            R E S T A R T
-                        </Button>
-                        <Button variant="contained" color="secondary">
-                            C H E C K
-                        </Button>
+            <div className="screen">
+                <div className="game" /*onKeyDown={this.handleKeyDown} tabIndex={0}*/>
+                    <KeyDownListener
+                        onKeyDown={this.handleGlobalKeyDown}
+                        onMouseDown={this.handleGlobalMouseDown}
+                    />
+                    <div className="board">{this.renderSquares()}</div>
+                    <div className="button-box">
+                        <div className="button-box-top">
+                            <ControlButtons
+                                onClickMode={this.toggleNumpadMode.bind(this)}
+                                numpadMode={this.state.numpadMode}
+                            />
```

Comment:
onClickMode is where im trying to test the normal control buttons preventDefault thing